### ISSUE TITLE
undefined method `zero?' for {:exact_text=>"somestring"}:Hash` for wait_until_<element>_invisible(text: 'somestring')

### ISSUE THREAD

**Issue by matisnape** [natritmeyer/site_prism#260]
## Environment
- Operating System: Mac
- Capybara: 2.18, Site_Prism: 2.11 
- RSpec 3.7.0 (e2e tests without rails app)
- Chrome 66.0.3359.181, chromedriver=2.37.544337

## Expected Behavior
I expected Capybara Query options to work - https://github.com/natritmeyer/site_prism#methods-supporting-capybara-options

I have a method:
```
  def save_question
    submit_btn(text: 'Add').click
    wait_until_submit_btn_invisible(text: 'Add')
  end
```
`submit_btn(text: 'Add').click` works as intended. 
Basing on the doc linked above, I would assume that if this `@results_page.wait_until_<element_or_section_name>_invisible(text: 'Some ajaxy text disappears!')` works, then `wait_until_submit_btn_invisible(text: 'Add')` should work. 

## Actual Behavior

`wait_until_submit_btn_invisible(text: 'Add')`

raises the following error:
```
 NoMethodError:
       undefined method `zero?' for {:exact_text=>"Add"}:Hash
     # /Users/anks/.rvm/gems/ruby-2.5.1/gems/site_prism-2.11/lib/site_prism/element_container.rb:156:in `block (2 levels) in create_invisibility_waiter'
     # ./support/sections/jobs/screening_questions_section.rb:52:in `save_question'
     # ./spec/jobs/screening_questions_spec.rb:42:in `block (4 levels) in <top (required)>'
     # /Users/anks/.rvm/gems/ruby-2.5.1/gems/rspec-retry-0.5.7/lib/rspec/retry.rb:115:in `block in run'
     # /Users/anks/.rvm/gems/ruby-2.5.1/gems/rspec-retry-0.5.7/lib/rspec/retry.rb:104:in `loop'
     # /Users/anks/.rvm/gems/ruby-2.5.1/gems/rspec-retry-0.5.7/lib/rspec/retry.rb:104:in `run'
     # /Users/anks/.rvm/gems/ruby-2.5.1/gems/rspec-retry-0.5.7/lib/rspec_ext/rspec_ext.rb:12:in `run_with_retry'
     # /Users/anks/.rvm/gems/ruby-2.5.1/gems/rspec-retry-0.5.7/lib/rspec/retry.rb:33:in `block (2 levels) in setup'
```

If I use simply `wait_until_submit_btn_invisible` without the query option, it works without error.

Also, look at this:
```
[3] pry(#<JobsScreeningQuestionsSection>)> has_no_submit_btn?(text: 'Add')
=> true
[4] pry(#<JobsScreeningQuestionsSection>)> wait_until_submit_btn_invisible(text: 'Add')
NoMethodError: undefined method `zero?' for {:text=>"Add"}:Hash
from /Users/anks/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/timeout.rb:76:in `timeout'
```

Not sure if this is an error or am I just using the query option incorrectly.

**Comment by luke-hill** [comment]
We have seen similar issues to this occurring due to frame usage. Can you attach a small sample page we can re-run this test against.

We have unit / feature tests for this specific use case which run successfully. So it's likely there could be some frame interfering.

Could you also attach your POM file, so we can see how your defining your selectors.

**Comment by matisnape** [comment]
@luke-hill I've put everything here https://gist.github.com/matisnape/a5480e83af17a89d82a4488b97480280

This page is built with Angular components, if that's helpful. 
Let me know if you need more details.

**Comment by luke-hill** [comment]
There are some tickets in which will fix some issues and possibly this.

However should these not work, are you able to (Not yet), produce a smaller more concise example. I logged in to that webpage and there is way too much going on. Ideally we'd want something in a single webpage, with a small example. See http://sscce.org/ for tips.

**Comment by luke-hill** [comment]
If the above is proving problematic you can see if using the latest uncut master works for you.

Amend your gemfile to point to our `master` branch. And that should be enough to give it a trial. Note that this is still unreleased, so could have some unknown issues.

**Comment by luke-hill** [comment]
Closing as fixed in 12478a4170d44521da6eb27921909313e75ec5c7
